### ISSUE TITLE
Statement parser doesn't break on hyphen

### ISSUE THREAD

**Issue by marekhanzlik** [mysql-net/MySqlConnector#563]
Hi i have some problems when switching from Mysql.Data to MySqlConnector

`var insertProductCmd = new MySqlCommand($"INSERT INTO {Tables.PlayerProductsTable} (product_id, player_id, product_count) VALUES (@productID, @playerID, 0) " +
                                                            $"ON DUPLICATE KEY " +
                                                            $"UPDATE product_count=GREATEST(0, product_count-@productCount)", conn, transaction);`

This command throws an `product_count`  cannot be null
This doesnt happen on MySql.Data, what am i missing?

Second problem is #562

**Comment by bgrainger** [comment]
Please provide a more complete sample (e.g., that shows parameter values, full exception message & call stack).

**Comment by marekhanzlik** [comment]
Whole command code
![Whole Command Code](https://i.imgur.com/jn3WA0R.png)

Exception `MySql.Data.MySqlClient.MySqlException: 'Column 'product_count' cannot be null'`
Parameter values are:
`@productID = 1`
`@playerID = 597`
`@productCount = 11`

Values in database
![Database values](https://i.imgur.com/Vwi7wrd.png)

**Comment by bgrainger** [comment]
This is a bug in MySqlConnector's parameter interpolation logic. You can work around it for now by changing your statement to the following (adding spaces around the minus sign):

```csharp
$"UPDATE product_count=GREATEST(0, product_count - @productCount)"
```

**Comment by marekhanzlik** [comment]
Thanks for info, i'll have to wait for fix instead of using workaround, this can show up on so many places in project it's not worth the risk of missing something

**Comment by bgrainger** [comment]
Fixed in [0.45.1](https://github.com/mysql-net/MySqlConnector/releases/tag/0.45.1).

**Comment by marekhanzlik** [comment]
Thank you very much 😊
